ert_shared/libres_facade.py

```diff
@@ -81,3 +83,120 @@ def select_or_create_new_case(self, case_name):
         if self.get_current_case_name() != case_name:
             fs = self._enkf_main.getEnkfFsManager().getFileSystem(case_name)
             self._enkf_main.getEnkfFsManager().switchFileSystem(fs)
+
+    def cases(self):
+        return self._enkf_main.getEnkfFsManager().getCaseList()
+
+    def is_case_hidden(self, case):
+        return self._enkf_main.getEnkfFsManager().isCaseHidden(case)
+
+    def case_has_data(self, case):
+        return self._enkf_main.getEnkfFsManager().caseHasData(case)
+
+    def is_case_running(self, case):
+        return self._enkf_main.getEnkfFsManager().isCaseRunning(case)
+
+    def all_data_type_keys(self):
+        return self._enkf_main.getKeyManager().allDataTypeKeys()
+
+    def observation_keys(self, key):
+        if self._enkf_main.getKeyManager().isGenDataKey(key):
+            key_parts = key.split("@")
+            key = key_parts[0]
+            if len(key_parts) > 1:
+                report_step = int(key_parts[1])
+            else:
+                report_step = 0
+
+            obs_key = GenDataObservationCollector.getObservationKeyForDataKey(self._enkf_main, key, report_step)
+            if obs_key is not None:
+                return [obs_key]
+            else:
+                return []
+        elif self._enkf_main.getKeyManager().isSummaryKey(key):
+            return [str(k) for k in self._enkf_main.ensembleConfig().getNode(key).getObservationKeys()]
+        else:
+            return []
+
+    def gather_gen_kw_data(self, case, key):
+        """ :rtype: pandas.DataFrame """
+        data = GenKwCollector.loadAllGenKwData(self._enkf_main, case, [key])
+        if key in data:
+            return data[key].dropna()
```

Comment:
This returns a Pandas.Series not a Pandas.DataFrame .

Reply:
You are correct. I have simply copied the code from the old PlotDataGatherer. Should I add the correct annotation or remove it?

Reply:
I think a dataframe is reasonable. You could just change to `data[key].to_frame().dropna()`